pulp_deb/tests/functional/api/test_crud_remotes.py

```diff
@@ -1,210 +1,157 @@
-
-
-class RemoteDownloadPolicyTestCase(unittest.TestCase):
-    """Verify download policy behavior for valid and invalid values.
 
-    In Pulp 3, there are are different download policies.
+def test_create_remote_repository_with_same_name(
+    apt_remote_api, remote_repository, verbose_remote_data
+):
+    """Verify whether it is possible to create a remote repository with the same
+    name.
+    """
+    body = deepcopy(verbose_remote_data)
+    body["name"] = remote_repository.name
+    with pytest.raises(ApiException):
+        apt_remote_api.create(body)
+
 
+def test_create_remote_repository_without_url(apt_remote_api, verbose_remote_data):
+    """Verify whether it is possible to create a remote repository without an URL.
+    This test targets the following issues:
+    * `Pulp #3395 <https://pulp.plan.io/issues/3395>`_
+    * `Pulp Smash #984 <https://github.com/pulp/pulp-smash/issues/984>`_
+    """
+    body = deepcopy(verbose_remote_data)
+    del body["url"]
+    with pytest.raises(ApiException):
+        apt_remote_api.create(body)
+
+
+def test_read_remote_by_href(apt_remote_api, remote_repository):
+    """Verify whether it is possible to read a remote repository by its href."""
+    href = apt_remote_api.read(remote_repository.pulp_href)
+    for key, val in remote_repository.to_dict().items():
+        assert href.to_dict()[key] == val
+
+
+def test_read_remote_repository_by_name(apt_remote_api, remote_repository):
+    """Verify whether it is possible to read a remote repository by its name."""
+    page = apt_remote_api.list(name=remote_repository.name)
+    assert len(page.results) == 1
+    for key, val in remote_repository.to_dict().items():
+        assert page.results[0].to_dict()[key] == val
+
+
+def test_update_remote_repository_with_patch(apt_remote_api, remote_repository):
+    """Test if a remote repository is updated by PATCH."""
+    body = _gen_verbose_remote()
+    response = apt_remote_api.partial_update(remote_repository.pulp_href, body)
+    monitor_task(response.task)
+    repo = apt_remote_api.read(remote_repository.pulp_href)
+    for key, val in body.items():
+        if key != "username" and key != "password":
+            assert repo.to_dict()[key] == val
+
+
+def test_update_remote_repository_with_put(apt_remote_api, remote_repository):
+    """Test if a remote repository is updated by PUT."""
+    body = _gen_verbose_remote()
+    response = apt_remote_api.update(remote_repository.pulp_href, body)
+    monitor_task(response.task)
+    repo = apt_remote_api.read(remote_repository.pulp_href)
+    for key, val in body.items():
+        if key != "username" and key != "password":
+            assert repo.to_dict()[key] == val
+
+
+def test_delete_remote_repository(apt_remote_api, remote_repository):
+    """Test if a remote repository can be deleted."""
+    response = apt_remote_api.delete(remote_repository.pulp_href)
+    monitor_task(response.task)
+    with pytest.raises(ApiException):
```

Comment:
Sometimes I use 
```suggestion
    with pytest.raises(ApiException) as exc:
```

And then below the with context manager I can still `assert` on it.